dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -530,6 +534,17 @@ private static Set<Author> getAllAuthors(Entry entry) {
         return allAuthors;
     }
 
+    private static Doi getSelectedConceptDoi(Entry<?, ?> entry) {
+        if (!entry.getConceptDois().isEmpty()) {
+            Doi selectedDoi = entry.getConceptDois().get(entry.getDoiSelection());
+            if (selectedDoi == null) {
+                selectedDoi = entry.getDefaultConceptDoi();
+            }
+            return selectedDoi;
+        }
+        return null;
+    }
```

Comment:
Could put this method in `Entry` instead, it seems like a helper ala `Entry.getDefaultConceptDoi`.